src/seedu/addressbook/storage/jaxb/AdaptedPerson.java

```diff
@@ -60,7 +60,7 @@ public AdaptedPerson(ReadOnlyPerson source) {
 
         address = new AdaptedContactDetail();
         address.isPrivate = source.getAddress().isPrivate();
-        address.value = source.getAddress().value;
+        address.value = source.getAddress().toString();
```

Comment:
i don't see your update to `toString()` method, hence I am assuming you will return the input address as is in `toString`. if this is the case, you will understand that the processed address stored in 4 objects and retrieved later may have inconsistencies in the format,